woocommerce/woocommerce-ios
# [Product Bundles] Add "Bundled products" row to product details

Part of: #8954


## Description


This adds a (non-actionable for now) "Bundled products" row to the product details screen for Product Bundles (custom `bundle` product type), behind a feature flag.

For now, the row shows fake data. This will be replaced with real data once the work is done in the other layers.

### Changes

* Adds a `productBundles` feature flag.
* Adds the "Bundled products" row view model to `DefaultProductFormTableViewModel`.
* Defines the rows to show for Product Bundles in `ProductFormActionsFactory`. These are the same as before (for non-core products) with the addition of the "Bundled products" row.
* Adds the new row to the Product Form (VC, table data source, etc.).
* Adds unit tests.

## Testing instructions


### Prerequisites

1. Install and activate the [Product Bundles extension](https://woocommerce.com/products/product-bundles/) on your store.
2. Create at least one product with the Product Bundle type.

### To test

1. Build and run the app.
2. Go to the Products tab.
3. Open a Product Bundle and confirm the new "Bundled products" row appears as expected.

## Screenshots


Before/Feature flag disabled|Feature flag enabled
-|-
![Simulator Screen Shot - iPhone 14 Pro - 2023-03-03 at 12 56 10](https://user-images.githubusercontent.com/8658164/222733847-91497a7f-d031-476e-92c9-216b17bd3868.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-03 at 12 53 39](https://user-images.githubusercontent.com/8658164/222733841-85d95aa6-bea1-4f73-bfdf-d623e92a40bd.png)


---
